### PR TITLE
RDoc-1079 Document session.Query.WhereRegex

### PR DESCRIPTION
**Pull request by ml054**

### CODE REVIEW

File: Documentation/4.0/Raven.Documentation.Pages/client-api/session/querying/how-to-use-regex.markdown [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+
+To return only documents that match regular expression, use the `Regex` method which enables RavenDB to perform server-side pattern matching queries. 
+
+## Example
+
+{CODE-TABS}
+{CODE-TAB:csharp:Sync regex_1@ClientApi\Session\Querying\HowToUseRegex.cs /}
+{CODE-TAB:csharp:Async regex_1_async@ClientApi\Session\Querying\HowToUseRegex.cs /}
+{CODE-TAB-BLOCK:csharp:RQL}
+from Users
```
Comment (ppekrol) [inline Documentation/4.0/Raven.Documentation.Pages/client-api/session/querying/how-to-use-regex.markdown]: Let's use Northwind samples

File: Documentation/4.0/Samples/csharp/Raven.Documentation.Samples/ClientApi/Session/Querying/HowToUseRegex.cs [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+        {
+            using (var store = new DocumentStore())
+            {
+                using (var session = store.OpenSession())
+                {
+                    #region regex_1
+                    // loads all users, which first name
+                    // starts with 'A' or 'J'
+                    List<User> users = session
+                        .Query<User>()
```
Comment (ppekrol) [inline Documentation/4.0/Samples/csharp/Raven.Documentation.Samples/ClientApi/Session/Querying/HowToUseRegex.cs]: Let's use Northwind sample here e.g. Product

File: Documentation/4.0/Raven.Documentation.Pages/client-api/session/querying/how-to-use-regex.markdown [hunk 3]
```diff
@@ -0,0 +1,15 @@
+# Session : Querying : How to Use Regexp
+
+To return only documents that match regular expression, use the `Regex` method which enables RavenDB to perform server-side pattern matching queries. 
+
```
Comment (ppekrol) [inline Documentation/4.0/Raven.Documentation.Pages/client-api/session/querying/how-to-use-regex.markdown]: Maybe add a note that pattern must be .NET compatible?

File: Documentation/4.0/Raven.Documentation.Pages/client-api/session/querying/how-to-use-regex.markdown [hunk 4]
```diff
@@ -0,0 +1,15 @@
+# Session : Querying : How to Use Regexp
```
Comment (ppekrol) [inline Documentation/4.0/Raven.Documentation.Pages/client-api/session/querying/how-to-use-regex.markdown]: typo: How to Use Regex

Comment (ml054) [reply Documentation/4.0/Raven.Documentation.Pages/client-api/session/querying/how-to-use-regex.markdown]: done

File: Documentation/4.0/Raven.Documentation.Pages/client-api/session/querying/how-to-use-regex.markdown (hunk 1 above)
Comment (ml054) [reply Documentation/4.0/Raven.Documentation.Pages/client-api/session/querying/how-to-use-regex.markdown]: done

File: Documentation/4.0/Samples/csharp/Raven.Documentation.Samples/ClientApi/Session/Querying/HowToUseRegex.cs (hunk 2 above)
Comment (ml054) [reply Documentation/4.0/Samples/csharp/Raven.Documentation.Samples/ClientApi/Session/Querying/HowToUseRegex.cs]: done

File: Documentation/4.0/Raven.Documentation.Pages/client-api/session/querying/how-to-use-regex.markdown (hunk 3 above)
Comment (ml054) [reply Documentation/4.0/Raven.Documentation.Pages/client-api/session/querying/how-to-use-regex.markdown]: done

**Comment by ml054** [comment]
@ppekrol Please recheck

**Pull request merged**